home/pi/auto_run.sh

```diff
@@ -548,6 +566,85 @@ then
     rm ~/first_run
 fi
 
+# Matrix Voice Hat Setup
+if [ -f ~/setup_matrix ]
+then
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo ""
+        echo "========================================================================="
+        echo "Setting up Matrix Voice Hat. This will install the matrixio-kernel-modules and pulseaudio"
+        echo "This will require 3 reboots"
+        echo "Press any key to continue..."
+        read -N1 -s anykey
+    else
+        echo "Press any key to continue setting up Matrix Voice HAT"
+        read -N1 -s anykey
+    fi
+
+    if [ ! -f matrix_setup_state.txt ]
+    then
+        echo "Add Matrix repo and key and Update and install packages"
+        # add repo
+        curl https://apt.matrix.one/doc/apt-key.gpg | sudo apt-key add -
+        echo "deb https://apt.matrix.one/raspbian $(lsb_release -sc) main" | sudo tee /etc/apt/sources.list.d/matrixlabs.list
+        sudo apt-get update -y
+        sudo apt-get upgrade -y
+
+        echo "stage-1" > matrix_setup_state.txt
+        echo "Needs to reboot in case of kernel updates"
+        read -p "Press enter to continue reboot"
+        sudo reboot
+    else
+        matrix_setup_state=$( cat matrix_setup_state.txt)
+    fi
+
+    if [ $matrix_setup_state == "stage-1" ]
+    then
+        echo "installing matcixio-kernel-modules"
+        sudo apt install matrixio-kernel-modules -y
+
+        echo "installing pulseaudio"
+        sudo apt-get install pulseaudio -y
+        
+        echo "Need to reboot after installing matcixio-kernel-modules and pulseaudio"
```

Comment:
echo "Rebooting to apply audio subsystem changes, the installation will continue afterwards."